ignite/contrib/handlers.py

```diff
@@ -0,0 +1,87 @@
+try:
+    from tqdm import tqdm
+except ImportError:
+    raise RuntimeError("This contrib module requires tqdm to be installed")
+
+from ignite.engine import Events
+
+
+class ProgressBar:
+    """
+    TQDM progress bar handler to log training progress and computed metrics
+
+    Args:
+        engine: ignite.Engine object
+        loader: iterable or dataloader object
+        output_transform: transform a function that transforms engine.state.output
+                into a dictionary of format {name: value}
+        mode (str): 'iteration' or 'epoch' (default=epoch)
+        log_frequency (int): frequency of which the metrics information is displayed (default=1)
+
+    Example:
+        (...)
+        pbar = ProgressBar(trainer, train_loader, output_transform=lambda x: {'loss': x})
+        trainer.add_handler(Events.ITERATION_COMPLETED, pbar)
+    """
+
+    def __init__(self, engine, loader, output_transform=lambda x: x, mode='epoch', log_frequency=1):
+        self.num_iterations = len(loader)
+        self.metrics = {}
+        self.alpha = 0.98
+        self.output_transform = output_transform
+        self.pbar = None
+        self.mode = mode
+        self.log_frequency = log_frequency
+
+        assert log_frequency >= 1, 'log_frequency must be positive'
+        assert mode in {'iteration', 'epoch'}, \
+            'incompatible mode {}, accepted modes {}'.format(mode, {'iteration', 'epoch'})
+
+        log_event = Events.EPOCH_COMPLETED if mode == 'epoch' else Events.ITERATION_COMPLETED
+
+        engine.add_event_handler(Events.EPOCH_STARTED, self._reset)
+        engine.add_event_handler(Events.EPOCH_COMPLETED, self._close)
+        engine.add_event_handler(log_event, self._log_message)
```

Comment:
I think you have a collision with `engine.add_event_handler(Events.EPOCH_COMPLETED, self._close)`

Reply:
why? the progress bar is still going to display the information, but it will flush it
I've tested it and it works well

Reply:
@miguelvr maybe you are right, just to be sure, did you tested in mode `epoch`?

Reply:
I've tested both, but I can add another unit test

Reply:
Okay, let me play with this handler too :)